contracts/igp-gas-oracle/src/contract_test.rs

```diff
@@ -0,0 +1,229 @@
+
+    let oracle_config = oracle.get_config()?;
+    assert_eq!(oracle_config.owner, deployer.to_string());
+    assert_eq!(oracle_config.pending_owner, None);
+
+    // test init transfer ownership
+    oracle.init_ownership_transfer(&deployer, &next_owner)?;
+
+    let oracle_config = oracle.get_config()?;
+    assert_eq!(oracle_config.owner, deployer.to_string());
+    assert_eq!(oracle_config.pending_owner, Some(next_owner.to_string()));
+
+    // test revoke transfer ownership
+    oracle.revoke_ownership_transfer(&deployer)?;
+
+    let oracle_config = oracle.get_config()?;
+    assert_eq!(oracle_config.owner, deployer.to_string());
+    assert_eq!(oracle_config.pending_owner, None);
+
+    // test claim transfer ownership
+    oracle.init_ownership_transfer(&deployer, &next_owner)?;
+    oracle.claim_ownership(&next_owner)?;
+
+    let oracle_config = oracle.get_config()?;
+    assert_eq!(oracle_config.owner, next_owner.to_string());
+    assert_eq!(oracle_config.pending_owner, None);
+
+    Ok(())
+}
+
+#[test]
+fn test_gas_data() -> anyhow::Result<()> {
+    let mut app = App::default();
+
+    let deployer = Addr::unchecked("deployer");
+
+    let mut oracle = setup_igp_gas_oracle(&mut app, &deployer)?;
+
+    // test single
+    let gas_config = RemoteGasDataConfig {
+        remote_domain: 1u32,
+        token_exchange_rate: Uint128::new(120921),
+        gas_price: Uint128::new(9120321),
+    };
+
+    oracle.set_remote_gas_data(&deployer, gas_config.clone())?;
+
+    let ret = oracle.get_exchange_rate_and_gas_price(gas_config.remote_domain)?;
+    assert_eq!(ret.exchange_rate, gas_config.token_exchange_rate);
+    assert_eq!(ret.gas_price, gas_config.gas_price);
+
+    // test multi
+    let gas_config = RemoteGasDataConfig {
+        remote_domain: 2u32,
+        token_exchange_rate: Uint128::new(120921),
+        gas_price: Uint128::new(9120321),
+    };
+
+    oracle.set_remote_gas_data_configs(&deployer, vec![gas_config.clone()])?;
+
+    let ret = oracle.get_exchange_rate_and_gas_price(gas_config.remote_domain)?;
+    assert_eq!(ret.exchange_rate, gas_config.token_exchange_rate);
+    assert_eq!(ret.gas_price, gas_config.gas_price);
+
+    Ok(())
+}
+
+mod test_execute_condition {
+
+    use super::*;
+
+    #[test]
+    fn test_init_ownership_transfer() -> anyhow::Result<()> {
+        let mut app = App::default();
+
+        let deployer = Addr::unchecked("deployer");
+        let next_owner = Addr::unchecked("next_owner");
+
+        let mut oracle = setup_igp_gas_oracle(&mut app, &deployer)?;
+
+        // ok
+        oracle.init_ownership_transfer(&deployer, &next_owner)?;
+
+        // fail - sender is not owner
+        oracle
+            .init_ownership_transfer(&next_owner, &deployer)
+            .unwrap_err();
+
+        // fail - pending_owner is not empty
+        oracle
+            .init_ownership_transfer(&deployer, &next_owner)
+            .unwrap_err();
```

Comment:
This test can only check the result is err, not suitable error came out.

Reply:
Yes will replace this to compare string